x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/apikey/UpdateApiKeyRequest.java

```diff
@@ -7,98 +7,44 @@
 import org.elasticsearch.core.Nullable;
-import org.elasticsearch.xpack.core.security.action.role.RoleDescriptorRequestValidator;
 import org.elasticsearch.xpack.core.security.authz.RoleDescriptor;
-import org.elasticsearch.xpack.core.security.support.MetadataUtils;
 
 import java.io.IOException;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 
-import static org.elasticsearch.action.ValidateActions.addValidationError;
-
-public final class UpdateApiKeyRequest extends ActionRequest {
+public final class UpdateApiKeyRequest extends BaseUpdateApiKeyRequest {
+    public static UpdateApiKeyRequest usingApiKeyId(final String id) {
+        return new UpdateApiKeyRequest(id, null, null);
+    }
 
     private final String id;
-    @Nullable
-    private final Map<String, Object> metadata;
-    @Nullable
-    private final List<RoleDescriptor> roleDescriptors;
 
     public UpdateApiKeyRequest(
         final String id,
         @Nullable final List<RoleDescriptor> roleDescriptors,
         @Nullable final Map<String, Object> metadata
     ) {
+        super(roleDescriptors, metadata);
         this.id = Objects.requireNonNull(id, "API key ID must not be null");
-        this.roleDescriptors = roleDescriptors;
-        this.metadata = metadata;
     }
 
     public UpdateApiKeyRequest(StreamInput in) throws IOException {
         super(in);
-        this.roleDescriptors = in.readOptionalList(RoleDescriptor::new);
-        this.metadata = in.readMap();
         this.id = in.readString();
     }
 
-    @Override
-    public ActionRequestValidationException validate() {
-        ActionRequestValidationException validationException = null;
-        if (metadata != null && MetadataUtils.containsReservedMetadata(metadata)) {
-            validationException = addValidationError(
-                "API key metadata keys may not start with [" + MetadataUtils.RESERVED_PREFIX + "]",
-                validationException
-            );
-        }
-        if (roleDescriptors != null) {
-            for (RoleDescriptor roleDescriptor : roleDescriptors) {
-                validationException = RoleDescriptorRequestValidator.validate(roleDescriptor, validationException);
-            }
-        }
-        return validationException;
-    }
-
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-        out.writeOptionalCollection(roleDescriptors);
-        out.writeGenericMap(metadata);
         out.writeString(id);
     }
 
-    public static UpdateApiKeyRequest usingApiKeyId(String id) {
-        return new UpdateApiKeyRequest(id, null, null);
-    }
-
     public String getId() {
         return id;
     }
-
-    public Map<String, Object> getMetadata() {
-        return metadata;
-    }
-
-    public List<RoleDescriptor> getRoleDescriptors() {
-        return roleDescriptors;
-    }
-
-    @Override
-    public boolean equals(Object o) {
```

Comment:
This was only used in a single test. To avoid the headache around equality and inheritance, removing it.